frontend/src/pages/Metrics.jsx

```diff
@@ -0,0 +1,118 @@
+import React, { useEffect, useState } from 'react'
+import Layout from '../layouts/Layout.jsx'
+import { NavLink, useParams } from 'react-router-dom'
+import '../styles/pages/Metrics.css'
+import '@fortawesome/fontawesome-free/css/all.min.css';
+
+const MetricRaws = (props) => {
+    useEffect(() => {
+        const fetchMetrics = async () => {
+            const data = await props.getAllMetrics();
+            console.log("get data:: ", data);
+            props.setMetricData(data);
+            props.setLoading(false); 
+        };
+
+        fetchMetrics(); 
+    }, []);
+
+    if (props.loading) {
+        return <div>Loading...</div>; 
+    }
+
+    if (!props.metricData) {
+        return <div>No data available</div>;
+    }
+
+    return props.metricData.map((metric, index) => (
+        <tr key={index}>
+            <th scope="row">{index + 1}</th>
+            <td>{metric.name}</td>
+            <td>{metric.description}</td>
+            <td>{metric.type}</td>
+            <td>{metric.area.join(',')}</td>
+            <th>
+                <NavLink to={`/dashboard/metrics/edit/${metric.id}`} 
+                    className={(navData) => navData.isActive ? 'active' : ''}>
+                    <button className="action-btn mx-1">
+                        <i className="fa-solid fa-pen-to-square"></i>
+                    </button>
+                </NavLink>
+                <button className="action-btn mx-1" onClick={()=>props.handleDelete(metric.id)}>
+                    <i className="fa-solid fa-trash"></i>
+                </button>
+            </th>
+        </tr>
+    ))
+        
+};
+
+
+const Metrics = () => {
+    const [metricData, setMetricData] = useState(null);
+    const [loading, setLoading] = useState(true);
+
+    useEffect(() => {console.log("re-rendered")}, []);
+    const getAllMetrics = async () => {
```

Comment:
please refactor these two functions as well such that they are moved to the api folder

Reply:
Done